### PR TITLE
chore: fixes inaccurate detection of fake mongodb deployments - COMPASS-7045

### PR DESCRIPTION
**Pull request by himanshusinghs**
## Description
A follow up of https://github.com/mongodb-js/mongodb-build-info/pull/22, which brings updated logic for fake deployment detection to Compass.

### Checklist
- [x] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [x] Misc

## Types of changes

- [ ] *Backport Needed*
- [x] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/data-service/src/instance-detail-helper.ts [hunk 1]
```diff
@@ -108,12 +107,12 @@ export type InstanceDetails = {
 };
 
 export async function getInstance(
-  client: MongoClient
+  client: MongoClient,
+  uri: string
```
Comment (himanshusinghs) [inline packages/data-service/src/instance-detail-helper.ts:111]: Is there a better way for me to get the connection uri here? MongoClient is in scope but does not provide an accessor to fetch the connection string.

File: packages/data-service/test/helpers.ts [hunk 2]
```diff
@@ -92,5 +95,19 @@ export function createMongoClientMock({
...
+  const connectionString = hosts.reduce<string>(
+    (connectionString, { host, port }, index) => {
+      const hostWithPort = port ? `${host}:${port}` : host;
+      if (index === 0) {
+        return `${connectionString}://${hostWithPort}`;
+      }
+      return `${connectionString},${hostWithPort}`;
+    },
+    'mongodb'
+  );
```
Comment (addaleax) [inline packages/data-service/test/helpers.ts:107]: Do you just want to use the `ConnectionString` class from the `mongodb-connection-string-url` package here? Here, you'd technically have to worry about things like escaping characters in `host` and `port` as well (e.g. for IPv6 hosts)

File: packages/data-service/src/instance-detail-helper.ts (hunk 1 above)
Comment (addaleax) [reply packages/data-service/src/instance-detail-helper.ts:111]: I think `MongoClient`s only store the parsed connection string options, not the original string, unfortunately 😐

**Review by addaleax — APPROVED** [review]

File: packages/data-service/test/helpers.ts (hunk 2 above)
Comment (himanshusinghs) [reply packages/data-service/test/helpers.ts:107]: yea that is right. I will make the adjustments.

**Pull request merged**